### PR TITLE
[TUN-3043] Setup for private podspec repo

### PR DESCRIPTION
**Pull request by azubala**
JIRA: https://observatory.atlassian.net/browse/TUN-3043

What's up:
- Updated .gitignore.
- Cherry pick from upstream: Add macOS support (0beb918)
- Excluded unnecessary platform dependencies.
- Added complied CPAProxy dependencies to the project.
- Added `fastlane` and `Jenkinsfile` for automatic pod release.

---
```
 -> CPAProxy (1.2.0)
    - WARN  | source: Git sources should specify a tag.
    - WARN  | xcodebuild:  /Users/aleksanderzubala/Developer/ynd-consult/Pods/CPAProxy/CPAProxy/CPAConfiguration.h:19:9: warning: parameter 'Path' not found in the function declaration [-Wdocumentation]
    - WARN  | xcodebuild:  /Users/aleksanderzubala/Developer/ynd-consult/Pods/CPAProxy/CPAProxy/CPAConfiguration.h:20:9: warning: parameter 'Path' not found in the function declaration [-Wdocumentation]
    - WARN  | xcodebuild:  /Users/aleksanderzubala/Developer/ynd-consult/Pods/CPAProxy/CPAProxy/CPAConfiguration.h:21:9: warning: parameter 'Path' not found in the function declaration [-Wdocumentation]
    - WARN  | xcodebuild:  /Users/aleksanderzubala/Developer/ynd-consult/Pods/CPAProxy/CPAProxy/CPAConfiguration.h:33:9: warning: parameter 'Path' not found in the function declaration [-Wdocumentation]
    - NOTE  | xcodebuild:  /Users/aleksanderzubala/Developer/ynd-consult/Pods/CPAProxy/CPAProxy/CPAConfiguration.h:33:9: note: did you mean 'torDataDirectoryPath'?
    - WARN  | xcodebuild:  /Users/aleksanderzubala/Developer/ynd-consult/Pods/CPAProxy/CPAProxy/CPAProxyManager.h:94:9: warning: parameter 'success' not found in the function declaration [-Wdocumentation]
    - WARN  | xcodebuild:  /Users/aleksanderzubala/Developer/ynd-consult/Pods/CPAProxy/CPAProxy/CPAProxyManager.h:95:9: warning: parameter 'failure' not found in the function declaration [-Wdocumentation]
    - WARN  | xcodebuild:  /Users/aleksanderzubala/Developer/ynd-consult/Pods/CPAProxy/CPAProxy/CPAProxyManager.h:103:9: warning: parameter 'success' not found in the function declaration [-Wdocumentation]
    - WARN  | xcodebuild:  /Users/aleksanderzubala/Developer/ynd-consult/Pods/CPAProxy/CPAProxy/CPAProxyManager.h:104:9: warning: parameter 'failure' not found in the function declaration [-Wdocumentation]
    - WARN  | xcodebuild:  /Users/aleksanderzubala/Developer/ynd-consult/Pods/CPAProxy/CPAProxy/CPASocketManager.h:67:13: warning: empty paragraph passed to '@param' command [-Wdocumentation]
    - WARN  | xcodebuild:  /Users/aleksanderzubala/Developer/ynd-consult/Pods/CPAProxy/CPAProxy/CPASocketManager.h:74:9: warning: parameter 'string' not found in the function declaration [-Wdocumentation]
    - NOTE  | xcodebuild:  /Users/aleksanderzubala/Developer/ynd-consult/Pods/CPAProxy/CPAProxy/CPASocketManager.h:74:9: note: did you mean 'command'?
    - WARN  | xcodebuild:  /Users/aleksanderzubala/Developer/ynd-consult/Pods/CPAProxy/CPAProxy/CPASocketManager.h:75:9: warning: parameter 'encoding' not found in the function declaration [-Wdocumentation]
    - NOTE  | xcodebuild:  /Users/aleksanderzubala/Developer/ynd-consult/Pods/CPAProxy/CPAProxy/CPASocketManager.h:75:9: note: did you mean 'command'?
    - NOTE  | [iOS] xcodebuild:  ld: warning: could not create compact unwind for _cmult: stack subl instruction is too different from dwarf stack size
    - NOTE  | [iOS] xcodebuild:  ld: warning: could not create compact unwind for _fmonty: stack subl instruction is too different from dwarf stack size

CPAProxy passed validation.
```

### CODE REVIEW

File: CPAProxy.podspec [hunk 1]
```diff
@@ -1,23 +1,34 @@
...
   s.name            = "CPAProxy"
-  s.version         = "1.1.0"
+  s.version         = "1.2.0"
   s.summary         = "CPAProxy is a thin Objective-C wrapper around Tor."
   s.author          = "Claudiu-Vlad Ursache <claudiu.vlad.ursache@gmail.com>"
 
   s.homepage        = "https://github.com/ursachec/CPAProxy"
   s.license         = { :type => 'MIT', :file => 'LICENSE.md' }
-  s.source          = { :git => "https://github.com/chrisballinger/CPAProxy.git", :branch => "podspec"}
+  s.source          = { :git => "https://github.com/ursachec/CPAProxy.git", :branch => "master"}
```
Comment (azubala) [inline CPAProxy.podspec]: This should point to YND fork and not to `master` but tagged version -- I'll update that.

**Comment by azubala** [comment]
@damiankolakowski please review if the updated library works fine with macOS app.

**Review by lukewar — APPROVED** [review]
LGTM

**Comment by damiankolakowski** [comment]
All is good at macOS app. Let's merge.

**Comment by azubala** [comment]
@lukewar I've updated `Fastfile` and `gitignore` the same way we did for SocketRocket. Do we need to do something Jenkins side to create those master & PR builds?

**Review by lukewar — APPROVED** [review]

**Pull request merged**